### PR TITLE
Update source container build plugins with build_dir

### PR DESCRIPTION
**Pull request by tkdchen**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

This PR requires #1718 to be merged firstly.

### CODE REVIEW

File: atomic_reactor/dirs.py [hunk 1]
```diff
@@ -253,3 +253,18 @@ def for_all_platforms_copy(self, action: FileCreationFunc) -> List[Path]:
                     copyfile(src_file, dest)
 
         return the_new_files
+
+
+class SourceBuildDir(RootBuildDir):
```
Comment (chmeliik) [inline atomic_reactor/dirs.py:273]: This kinda sounds like a combination of the `Source` and `BuildDir` classes (which it isn't). Maybe rename it to `SourceContainersBuildDir`? Or maybe it doesn't need to be a subclass, having the attributes in RootBuildDir might be OK if they have "source_container" in the name

**Review by chmeliik — COMMENTED** [review]
I was going to ask if we are missing `bump_release` changes, but it turns out it doesn't touch the build directory at all for source containers.

The source container and binary container workflows really have almost nothing in common, huh :smile:

File: atomic_reactor/dirs.py [hunk 2]
```diff
@@ -260,11 +260,16 @@ class SourceBuildDir(RootBuildDir):
 
     def __init__(self, path: Path) -> None:
         super().__init__(path)
-        self._sources_dir: Optional[Path] = None
+        self._sources_dir: Path = self.path / "sources"
+        self._output_dir: Path = self.path / "output"
+        self.exported_squashed_image: Path = self.path / EXPORTED_SQUASHED_IMAGE_NAME
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/dirs.py:280]: exported image is output

also in RootBuildDir for binary builds you will need that for all arches

please add comments about sources & output ...........

building on top of that platform structure doesn't make much sense for source containers, but for consistency it is fine

File: atomic_reactor/plugins/post_compress.py [hunk 3]
```diff
@@ -52,14 +52,12 @@ def __init__(self, workflow, load_exported_image=False, method='gzip'):
         self.uncompressed_size = 0
         self.source_build = bool(self.workflow.build_result.source_docker_archive)
 
-    def _compress_image_stream(self, stream):
-        outfile = os.path.join(self.workflow.source.workdir,
-                               EXPORTED_COMPRESSED_IMAGE_NAME_TEMPLATE)
+    def _compress_image_stream(self, stream) -> Path:
         if self.method == 'gzip':
-            outfile = outfile.format('gz')
+            outfile = self.workflow.build_dir.exported_compressed_image('gz')
```
Comment (rcerven) [inline atomic_reactor/plugins/post_compress.py:57]: so this is change which will work only for source builds

for container builds you will need it for all arches

File: atomic_reactor/dirs.py [hunk 4]
```diff
@@ -115,6 +129,21 @@ def __init__(self, path: Path) -> None:
             raise FileNotFoundError(f"Path {path} does not exist.")
         self.path = path
         self.platforms: Optional[List[str]] = None
+        self._source_container_output_dir: Path = self.path / "output"
```
Comment (chmeliik) [inline atomic_reactor/dirs.py:132]: Not needed

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

Just found one unnecessary attribute

File: atomic_reactor/dirs.py (hunk 4 above)
Comment (tkdchen) [reply atomic_reactor/dirs.py]: Oops. Removed.

**Review by rcerven — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]

**Pull request merged**